src/Monolog/Logger.php

```diff
@@ -387,7 +404,11 @@ public function addRecord(int $level, string $message, array $context = [], Date
             }
         } finally {
             if ($this->detectCycles) {
-                $this->logDepth--;
+                if (\PHP_VERSION_ID >= 80100 && $fiber = \Fiber::getCurrent()) {
```

Comment:
Kinda unfortunate that we get a hit here of getting the current fiber for every log created, even if not using fibers at all.. I don't know maybe that's got very little perf impact and it's fine?

Reply:
We can reduce it to a single call instead of two by storing the current fiber in a local variable. I think a single call there is a fine price to pay for it working by default, but I'm obviously biased.

Reply:
Yeah I mean this call could definitely be removed as $fiber is already set above, but my point was more avoiding both calls ;) But I do see the point about having it just work.

Seems like 100K calls to Fiber::getCurrent are done in 7/8ms in a quick test here, so I guess this is really a non-issue.

Reply:
Yes, apart from the typical function call overhead, it's very cheap, just a global lookup:

https://github.com/php/php-src/blob/396b2aab85ebab0a72206e74ce87eda75a4167fd/Zend/zend_fibers.c#L864-L875

What should I do about the PHPStan errors, seems like PHPStan doesn't know about fibers?

Reply:
Yeah we can ignore those in the phpstan config I guess, and report it to PHPStan if there is no issue for it yet.

Reply:
went ahead a reported a phpstan issue: https://github.com/phpstan/phpstan/issues/7843

Reply:
Thanks!

Reply:
I've added ignores and removed the duplicate call to `Fiber::getCurrent()`.